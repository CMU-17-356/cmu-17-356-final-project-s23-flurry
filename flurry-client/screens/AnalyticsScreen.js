import React from 'react';
import { View, Text, StyleSheet, SectionList } from 'react-native';
import DateTimeRangePicker from '../components/DateTimeRangePicker';
import DriverDropdown from '../components/DriverDropdown';
import SlipSeverityChart from '../components/SlipSeverityChart';
import { slipIncidents } from '../data/dummyData';

export default class AnalyticsScreen extends React.Component {
  handleDateRangeChange = (startDate, endDate) => {
    // Do something with the selected start and end dates
    //console.log('Selected date range:', startDate, endDate);
  };
  handleDriverSelect = (driverName) => {
    setSelectedDriver(driverName);
  };

  render() {
    return (
      <SectionList
        contentContainerStyle={{ alignItems: 'center', justifyContent: 'center'}}
        sections={[
          {
            title: 'Pick a date range for Analytics',
            data: [<DateTimeRangePicker onDateRangeChange={this.handleDateRangeChange} />],
          },
          {
            title: '',
            data: [<DriverDropdown onDriverSelect={this.handleDriverSelect} />],
          },
          {
            title: '',
            data: [<SlipSeverityChart data={slipIncidents} />],
          },
        ]}
        keyExtractor={(item, index) => index.toString()}
        renderItem={({ item }) => <View style={styles.item}>{item}</View>}
        renderSectionHeader={({ section: { title } }) => (
          <View style={styles.header}>
            <Text style={styles.title}>{title}</Text>
          </View>
        )}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 10,
    marginTop: 30,
  },
  item: {
    marginVertical: 10,
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold',
    marginVertical: 10,
    marginBottom: 10,
  },
});