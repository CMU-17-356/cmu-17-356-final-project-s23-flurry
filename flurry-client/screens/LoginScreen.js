import { createStackNavigator } from "@react-navigation/stack";
import Checkbox from "expo-checkbox";
import { Image } from "expo-image";
import * as ImageManipulator from "expo-image-manipulator";
import React, { Component } from "react";
import { Alert, Button, TextInput, View, StyleSheet, Text } from "react-native";

import FlurryIcon from "../assets/icon.png";

export default class LoginScreen extends Component {
  constructor(props) {
    super(props);

    this.state = {
      props,
      username: "",
      password: "",
      isLogin: true, // toggle login/signup
      name: "",
      company: "",
      isManager: false, // toggle driver/manager
    };
  }

  onLogin() {
    const { props, username, password, isLogin, name, company, isManager } =
      this.state;

    let formData;
    let apiCall;
    if (isLogin) {
      apiCall = "login";
      formData = {
        id: username,
        password,
      };
    } else {
      apiCall = "accounts";
      formData = {
        id: username,
        password,
        name,
        company_id: "c_" + company.toLowerCase(),
      };
    }

    const role = isManager ? "manager" : "driver";

    fetch("https://flurry-backend.fly.dev/api/" + apiCall + "?type=" + role, {
      method: "POST",
      headers: {
        "Content-type": "application/json; charset=UTF-8",
      },
      body: JSON.stringify(formData),
    })
      .then((res) => {
        const status = res["status"];
        if (status === 400) {
          Alert.alert("Invalid fields");
        } else if (status === 404) {
          Alert.alert("Invalid query param type");
        } else {
          if (isManager) {
            this.props.navigation.navigate("Analytics");
          } else {
            this.props.navigation.navigate("Driver Info");
          }
        }
      })
      .catch((error) => {
        console.log("There was a problem with the fetch operation:", error);
      });
  }

  render() {
    return (
      <View style={styles.container}>
        <Image
          source={FlurryIcon}
          contentFit="cover"
          transition={1000}
          style={styles.image}
        />
        {!this.state.isLogin && (
          <>
            <TextInput
              value={this.state.name}
              onChangeText={(name) => this.setState({ name })}
              placeholder={"Name"}
              style={styles.input}
            />
            <TextInput
              value={this.state.company}
              onChangeText={(company) => this.setState({ company })}
              placeholder={"Company"}
              style={styles.input}
            />
          </>
        )}
        <TextInput
          value={this.state.username}
          onChangeText={(username) => this.setState({ username })}
          placeholder={"Username"}
          style={styles.input}
        />
        <TextInput
          value={this.state.password}
          onChangeText={(password) => this.setState({ password })}
          placeholder={"Password"}
          secureTextEntry={true}
          style={styles.input}
        />
        <View style={styles.checkboxWrapper}>
          <Checkbox
            style={styles.checkbox}
            value={this.state.isManager}
            onValueChange={() =>
              this.setState({ isManager: !this.state.isManager })
            }
            color={this.state.isManager ? "#007AFF" : undefined}
          />
          <Text style={styles.text}>Manager</Text>
        </View>

        <Button
          title={this.state.isLogin ? "Login" : "Sign up"}
          style={styles.input}
          onPress={this.onLogin.bind(this)}
        />
        <Button
          title={
            this.state.isLogin
              ? "Don't have an account? Sign up"
              : "Already have an account? Log in"
          }
          style={styles.alt}
          onPress={() => this.setState({ isLogin: !this.state.isLogin })}
          color="#888888"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white",
    height: "100%",
    width: "100%",
    paddingTop: "5%",
  },
  input: {
    width: 200,
    height: 44,
    padding: 10,
    borderWidth: 1,
    borderColor: "black",
    marginBottom: 10,
  },
  alt: {
    width: 200,
    height: 44,
    padding: 10,
    borderWidth: 1,
    borderColor: "black",
    marginBottom: 10,
  },
  image: {
    width: 100,
    height: 100,
    margin: 10,
  },
  checkbox: {
    margin: 8,
  },
  text: {
    lineHeight: 37,
  },
  checkboxWrapper: {
    display: "flex",
    flexDirection: "row",
    alignContent: "center",
    paddingBottom: 5,
  },
});
