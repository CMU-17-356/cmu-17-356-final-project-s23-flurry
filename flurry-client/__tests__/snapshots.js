import renderer from "react-test-renderer";

import LoginScreen from "../Screens/LoginScreen";
import AnalyticsScreen from "../screens/AnalyticsScreen";
import MapScreen from "../screens/MapScreen";

test("login screen rendering", async () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("map screen rendering", async () => {
  const tree = renderer.create(<MapScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("analytics screen rendering", async () => {
  const tree = renderer.create(<AnalyticsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
