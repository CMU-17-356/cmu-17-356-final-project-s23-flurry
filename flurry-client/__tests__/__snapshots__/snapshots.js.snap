// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analytics screen rendering 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "justifyContent": "center",
    }
  }
  data={
    [
      {
        "data": [
          <DateTimeRangePicker
            onDateRangeChange={[Function]}
          />,
        ],
        "title": "",
      },
      {
        "data": [
          <DriverDropdown
            drivers={[]}
            onSelectDriver={[Function]}
            slips={[]}
          />,
        ],
        "title": "",
      },
      {
        "data": [
          <SlipSeverityChart
            data={[]}
          />,
        ],
        "title": "",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    [
      0,
      3,
      6,
    ]
  }
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View>
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginVertical": 10,
            }
          }
        />
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "marginVertical": 10,
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "paddingVertical": 15,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#007aff",
                    "fontSize": 16,
                  }
                }
              >
                Start Date: 
                Sat Jan 01 2000
              </Text>
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              isVisible={false}
              onHide={[Function]}
              transparent={true}
              visible={false}
            />
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "paddingVertical": 15,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#007aff",
                    "fontSize": 16,
                  }
                }
              >
                Start Time: 
                12:00 AM
              </Text>
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              isVisible={false}
              onHide={[Function]}
              transparent={true}
              visible={false}
            />
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "paddingVertical": 15,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#007aff",
                    "fontSize": 16,
                  }
                }
              >
                End Date: 
                Wed May 03 2023
              </Text>
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              isVisible={false}
              onHide={[Function]}
              transparent={true}
              visible={false}
            />
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "paddingVertical": 15,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#007aff",
                    "fontSize": 16,
                  }
                }
              >
                End Time: 
                11:29 PM
              </Text>
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              isVisible={false}
              onHide={[Function]}
              transparent={true}
              visible={false}
            />
          </View>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View>
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginVertical": 10,
            }
          }
        />
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "marginVertical": 10,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View>
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginVertical": 10,
            }
          }
        />
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "marginVertical": 10,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "#000",
              "borderWidth": 1,
              "justifyContent": "center",
              "marginBottom": 10,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            Slip Severity Distribution
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "height": 220,
                  "padding": 0,
                  "width": 400,
                }
              }
            >
              <RNSVGSvgView
                bbHeight="220"
                bbWidth="400"
                focusable={false}
                height={220}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 220,
                      "width": 400,
                    },
                  ]
                }
                width={400}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGDefs>
                      <RNSVGLinearGradient
                        gradient={
                          [
                            0,
                            -1,
                            1,
                            -1,
                          ]
                        }
                        gradientTransform={null}
                        gradientUnits={1}
                        name="backgroundGradient"
                        x1="0"
                        x2="220"
                        y1="220"
                        y2="0"
                      />
                      <RNSVGLinearGradient
                        gradient={
                          [
                            0,
                            436207616,
                            1,
                            436207616,
                          ]
                        }
                        gradientTransform={null}
                        gradientUnits={1}
                        name="fillShadowGradientFrom"
                        x1="0"
                        x2="0"
                        y1="0"
                        y2="220"
                      />
                    </RNSVGDefs>
                  </RNSVGGroup>
                  <RNSVGRect
                    fill={
                      {
                        "payload": 0,
                        "type": 0,
                      }
                    }
                    height="220"
                    propList={
                      [
                        "fill",
                      ]
                    }
                    rx="0"
                    ry="0"
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    matrix={
                      [
                        1,
                        0,
                        -0,
                        1,
                        115,
                        110,
                      ]
                    }
                  >
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGPath
                        d="M 10 -88 A 88 88 0 0 1 10 -88 L 10 0 A 0 0 0 0 0 10 0 Z "
                        fill={
                          {
                            "payload": 4281257073,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4281257073,
                            "type": 0,
                          }
                        }
                        height="16px"
                        propList={
                          [
                            "fill",
                          ]
                        }
                        rx="8"
                        ry="8"
                        width="16px"
                        x="136"
                        y="-76"
                      />
                      <RNSVGText
                        content={null}
                        dx={[]}
                        dy={[]}
                        fill={
                          {
                            "payload": 4281257073,
                            "type": 0,
                          }
                        }
                        font={{}}
                        propList={
                          [
                            "fill",
                          ]
                        }
                        rotate={[]}
                        x={
                          [
                            "160",
                          ]
                        }
                        y={
                          [
                            "-64",
                          ]
                        }
                      >
                        <RNSVGTSpan
                          content="0 Severity <= 30"
                          dx={[]}
                          dy={[]}
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                          font={{}}
                          rotate={[]}
                          x={[]}
                          y={[]}
                        />
                      </RNSVGText>
                    </RNSVGGroup>
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGPath
                        d="M 10 -88 A 88 88 0 0 1 10 -88 L 10 0 A 0 0 0 0 0 10 0 Z "
                        fill={
                          {
                            "payload": 4281637083,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4281637083,
                            "type": 0,
                          }
                        }
                        height="16px"
                        propList={
                          [
                            "fill",
                          ]
                        }
                        rx="8"
                        ry="8"
                        width="16px"
                        x="136"
                        y="-32"
                      />
                      <RNSVGText
                        content={null}
                        dx={[]}
                        dy={[]}
                        fill={
                          {
                            "payload": 4281637083,
                            "type": 0,
                          }
                        }
                        font={{}}
                        propList={
                          [
                            "fill",
                          ]
                        }
                        rotate={[]}
                        x={
                          [
                            "160",
                          ]
                        }
                        y={
                          [
                            "-20",
                          ]
                        }
                      >
                        <RNSVGTSpan
                          content="0 Severity <= 50"
                          dx={[]}
                          dy={[]}
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                          font={{}}
                          rotate={[]}
                          x={[]}
                          y={[]}
                        />
                      </RNSVGText>
                    </RNSVGGroup>
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGPath
                        d="M 10 -88 A 88 88 0 0 1 10 -88 L 10 0 A 0 0 0 0 0 10 0 Z "
                        fill={
                          {
                            "payload": 4294034447,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4294034447,
                            "type": 0,
                          }
                        }
                        height="16px"
                        propList={
                          [
                            "fill",
                          ]
                        }
                        rx="8"
                        ry="8"
                        width="16px"
                        x="136"
                        y="12"
                      />
                      <RNSVGText
                        content={null}
                        dx={[]}
                        dy={[]}
                        fill={
                          {
                            "payload": 4294034447,
                            "type": 0,
                          }
                        }
                        font={{}}
                        propList={
                          [
                            "fill",
                          ]
                        }
                        rotate={[]}
                        x={
                          [
                            "160",
                          ]
                        }
                        y={
                          [
                            "24",
                          ]
                        }
                      >
                        <RNSVGTSpan
                          content="0 Severity <= 70"
                          dx={[]}
                          dy={[]}
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                          font={{}}
                          rotate={[]}
                          x={[]}
                          y={[]}
                        />
                      </RNSVGText>
                    </RNSVGGroup>
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGPath
                        d="M 10 -88 A 88 88 0 0 1 10 -88 L 10 0 A 0 0 0 0 0 10 0 Z "
                        fill={
                          {
                            "payload": 4293348412,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4293348412,
                            "type": 0,
                          }
                        }
                        height="16px"
                        propList={
                          [
                            "fill",
                          ]
                        }
                        rx="8"
                        ry="8"
                        width="16px"
                        x="136"
                        y="56"
                      />
                      <RNSVGText
                        content={null}
                        dx={[]}
                        dy={[]}
                        fill={
                          {
                            "payload": 4293348412,
                            "type": 0,
                          }
                        }
                        font={{}}
                        propList={
                          [
                            "fill",
                          ]
                        }
                        rotate={[]}
                        x={
                          [
                            "160",
                          ]
                        }
                        y={
                          [
                            "68",
                          ]
                        }
                      >
                        <RNSVGTSpan
                          content="0 Severity <= 100"
                          dx={[]}
                          dy={[]}
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                          font={{}}
                          rotate={[]}
                          x={[]}
                          y={[]}
                        />
                      </RNSVGText>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;

exports[`login screen rendering 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "5%",
      "width": "100%",
    }
  }
>
  <ViewManagerAdapter_ExpoImage
    height={100}
    margin={10}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    proxiedProperties={
      {
        "accessibilityLabel": undefined,
        "contentFit": "cover",
        "contentPosition": {
          "left": "50%",
          "top": "50%",
        },
        "placeholder": [],
        "source": [],
        "tintColor": undefined,
        "transition": {
          "duration": 1000,
        },
      }
    }
    style={
      {
        "height": 100,
        "margin": 10,
        "width": 100,
      }
    }
    width={100}
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Username"
    style={
      {
        "borderColor": "black",
        "borderWidth": 1,
        "height": 44,
        "marginBottom": 10,
        "padding": 10,
        "width": 200,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      {
        "borderColor": "black",
        "borderWidth": 1,
        "height": 44,
        "marginBottom": 10,
        "padding": 10,
        "width": 200,
      }
    }
    value=""
  />
  <View
    style={
      {
        "alignContent": "center",
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 5,
      }
    }
  >
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        {
          "busy": undefined,
          "checked": false,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderColor": "#657786",
            "borderRadius": 2,
            "borderWidth": 2,
            "height": 20,
            "width": 20,
          },
          {
            "margin": 8,
          },
          false,
          false,
          undefined,
          false,
        ]
      }
    />
    <Text
      style={
        {
          "lineHeight": 37,
        }
      }
    >
      Manager
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            {
              "color": "#888888",
            },
          ]
        }
      >
        Don't have an account? Sign up
      </Text>
    </View>
  </View>
</View>
`;

exports[`map screen rendering 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "height": "100%",
      "paddingTop": "5%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "width": "40%",
        }
      }
    >
      <Text>
        Start Date
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            12/31/2020
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            12:00 PM
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "width": "40%",
        }
      }
    >
      <Text>
        End Date
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            5/3/2023
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            11:29 PM
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
